pkg/template/controller/templateinstance_controller.go

```diff
@@ -51,9 +52,10 @@ const readinessTimeout = time.Hour
 // using its own service account, first verifying that the requester also has
 // permissions to instantiate.
 type TemplateInstanceController struct {
-	restmapper     meta.RESTMapper
-	config         *rest.Config
-	templateClient templateclient.Interface
+	dynamicRestMapper meta.RESTMapper
+	config            *rest.Config
```

Comment:
Technically this can be done with a codec against the dynamic client and can remove the kubectl dependencies this places in our controllers.  Doesn't have to be done here.  A TODO would be good.

Reply:
added todo